data_discovery_ai/agents/descriptionFormattingAgent.py

```diff
@@ -97,20 +98,43 @@ def take_action(self, title: str, abstract: str) -> str:
                             """
         response = None
         try:
-            client = OpenAI(api_key=self.openai_api_key)
-            completion = client.chat.completions.create(
-                model=self.model_config["model"],
-                messages=[
-                    {"role": "system", "content": system_prompt},
-                    {"role": "user", "content": input_text},
-                ],
-                temperature=self.model_config["temperature"],
-                max_tokens=self.model_config["max_tokens"],
-            )
-            response = completion.choices[0].message.content
+            # if use OpenAI model in production or staging
+            if self.model_config["model"] == "gpt-4o-mini":
+                client = OpenAI(api_key=self.openai_api_key)
+                completion = client.chat.completions.create(
+                    model=self.model_config["model"],
+                    messages=[
+                        {"role": "system", "content": system_prompt},
+                        {"role": "user", "content": input_text},
+                    ],
+                    temperature=self.model_config["temperature"],
+                    max_tokens=self.model_config["max_tokens"],
+                )
+                response = completion.choices[0].message.content
+            elif self.model_config["model"] == "llama3":
+                # in dev use free llama 3 model
+                system_prompt = """You are a marine science officer processing metadata records. Given a title and abstract of a metadata record, perform the following task:
```

Comment:
Consider use this prompt, which is faster

Process a metadata record's title and abstract. Reformat the abstract, keeping original text, using Markdown:
Lists: Each item on new line, start with -.

Headings: # H1, ## H2, ### H3, #### H4.

Bold: **text**.

Italics: *text*.

Links: URLs (www/http/https) as [text](URL).